pulsar-broker/src/main/java/org/apache/pulsar/broker/service/BrokerService.java

```diff
@@ -2009,4 +2025,49 @@ public ConfigField(Field field) {
             return Optional.empty();
         }
     }
+
+    private void enableTopicsAutoRead() {
+        topics.values().forEach(future -> {
+            if (future.isDone() && !future.isCompletedExceptionally()) {
+                try {
+                    future.get().ifPresent(Topic::enableProducerRead);
+                } catch (InterruptedException | ExecutionException e) {
+                   // no-op
+                }
+            }
+        });
+    }
+
+    @VisibleForTesting
+    boolean increasePublishBufferSizeAndCheckStopRead(int msgSize) {
+        if (maxMessagePublishBufferSize < 0) {
+            return false;
+        }
+        if (currentMessagePublishBufferSize.addAndGet(msgSize) >= maxMessagePublishBufferSize &&
```

Comment:
This would become a contention point across all the threads in the broker

Reply:
@merlimat Yes, this does increase competition, how about move `currentMessagePublishBufferSize` to ServerCnx and periodically sync them to `totalMessagePublishBufferSize` in BrokerService by a single thread?